src/node.cc

```diff
@@ -3047,6 +3054,20 @@ static void ParseArgs(int* argc,
     } else if (strcmp(arg, "--v8-options") == 0) {
       new_v8_argv[new_v8_argc] = "--help";
       new_v8_argc += 1;
+    } else if (strncmp(arg, "--cipher-list=", 14) == 0) {
```

Comment:
The intent on introducing --cipher-list in v0.10.x and v0.12.x is to avoid
the possibility of having to make these kinds of changes in the future. If,
for instance, another one of the default ciphers comes under scrutiny and
we have to switch it off, rather than having to bump the maintenance
release again we can simply tell folks to use the command line switch or
envar to override the list entirely and go from there.

On Tue, Apr 7, 2015 at 5:29 PM, Julien Gilli <notifications@github.com>
wrote:

> In src/node.cc
> <https://github.com/joyent/node/pull/14414#discussion_r27934008>:
>
> > @@ -3047,6 +3054,20 @@ static void ParseArgs(int* argc,
> >      } else if (strcmp(arg, "--v8-options") == 0) {
> >        new_v8_argv[new_v8_argc] = "--help";
> >        new_v8_argc += 1;
> > +    } else if (strncmp(arg, "--cipher-list=", 14) == 0) {
>
> Paraphrasing my previous comment
> <https://github.com/joyent/node/pull/14413#discussion_r27933665> for the
> corresponding PR targeted to v0.10.x:
>
> I'm wondering if we *need* to support a --cipher-list command line option
> in v0.12.x. Adding support for --enable-legacy-cipher-list already breaks
> the contract of not adding additional API in patch releases, and it seems
> sufficient to handle the use case of users not wanting to use the new
> default ciphers list.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joyent/node/pull/14414/files#r27934008>.
>